### PR TITLE
ovn 4.0 cleanups

### PR DESCRIPTION
**Pull request by pecameron**
Changes that permit ovn 4.0 to install and come up properly

Signed-off-by: Phil Cameron <pcameron@redhat.com>

### CODE REVIEW

**Comment by pecameron** [comment]
@shettyg @rajatchopra @squeed @dcbw PTAL

**Comment by shettyg** [comment]
@pecameron 

1. I don't see documentation change for READMEcontainer.md. The community dameonset running should follow those instructions right? Is it that, it does not need any change?

2. I see in the Makefile that a it automatically tries to push to dockerhub. That will likely fail as there is no username/password with the person that runs "make". What is the thought process here?

**Comment by pecameron** [comment]
@shettyg @rajatchopra @squeed @dcbw 
1. See dist/READMEopenshifttechpreview.md
We decided in a conversation that the *md files in ovn-kubernetes top level directory are out of date and need to be significantly revised. I have not undertaken to do that in this PR. I am not settled on how work going forward in ovn-kubernetes should be organized. On the openshift side the install in 4.0, when complete, will be completely automated by the network operator. (You can use that when you are working on Openshift). I don't know what we will be doing to support continued development and testing ovn-kubernetes. So in this interim period I copied the 3.11 dev preview install and edited it for 4.0 tech preview. I view this as a temporary solution until we decide how install will be handled going forward.

2. Makefile push to docker.io
First off the user is ovnkube. Rajat set this up you should have the password (if not, Rajat or I can email it to you). docker.io/ovnkube is the project repo that should be used for images that work with merged changes. The person that does the merge should also build and push the new image. 
The daemonset yaml files specify an image: docker.io/ovnkube/ovn-daemonset:latest that must be present and recent enough for the ansible install to just work and networking to come up. Toward that end whenever I change the image content or its use in the daemonsets things must be synced up again. I have been waiting for the PR to MERGE and then I build a new image and push it (the default make). I think that the person that does the merge should do that build and push. If you are doing development you need to build the image and push it to your personal repo and edit the daemonset yaml files with the new name (3 places in ovnkube.yaml and 2 places in ovnkube-master.yaml) At that point recreate the daemonsets and test.

**Comment by shettyg** [comment]
@pecameron 

I will try to reverse engineer the documentation needed for community daemonset from ansible scripts and openshift documentation. So right now, as I understand, READMEcontainer.md is stale - it used to work last week.

**Comment by shettyg** [comment]
Looks like you upgraded the community image today and it started working again for me.

**Comment by pecameron** [comment]
@shettyg I did upgrade the community image today. If we want to keep things working we need to build and push the image whenever a merge is done. All the code changes in controller.go and the ovnkube.sh script need to have an image built and pushed for the changes to become available to people that just install and run the software.

For the docs, the dist/ techpreview .md has a lot of info. Also dist/ansible playbook and dist/ansible/script/ovn-setup.sh have a lot. dist/yaml has the files needed to create the various pieces.
The existing docs seem to talk about an earlier version and has not been kept up to date. I have never set up a kubernetes cluster and I am not sure how to present it.

**Comment by shettyg** [comment]
@pecameron 

I will take care of the documentation for community daemonset. I have it working in vagrant. So I will get that in first and then make documentation changes to reflect how it is supposed to work. From there on, I will try to make sure that whenever your changes come in for dameonsets, I keep an eye on how it effects community dameonset.

**Comment by pecameron** [comment]
@shettyg is this PR ready to merge?

**Comment by shettyg** [comment]
/lgtm.

But I don't use ansible, so if no body else merges this in a day, I will.

**Comment by pecameron** [comment]
@dcbw @rajat @shettyg Please, Please, take a look and merge if OK. This is blocking another PR that I want to create. PTAL

File: dist/ansible/scripts/ovn-setup.sh [hunk 1]
```diff
@@ -66,6 +67,7 @@ wait_for_condition is_api_server_ready
 #    networkPluginName: cni
 #  serviceNetworkCIDR: 172.30.0.0/16
 
+# ocp 4.0 permits multiple CIDRs that include the subnet length
```
Comment (dcbw) [inline dist/ansible/scripts/ovn-setup.sh]: @pecameron it's not OCP4 that does this, it's ovn-kubernetes itself since Jacob added that support.

**Pull request merged**